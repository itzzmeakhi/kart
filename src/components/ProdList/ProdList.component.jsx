import React, { useEffect } from 'react';
import { connect } from 'react-redux';

import { getAllProductsStart } from './../../redux/products/products.actions';
import Product from './../Product/Product.component';

import './ProdList.styles.scss';

const ProdList = ({ fetchProducts, products }) => {
    useEffect(() => {
        fetchProducts();
    }, [fetchProducts])
    return (
        <div className = 'prodlist'>
            {products && products.map(prod => {
                return <Product
                            key = {prod.prodId}
                            prodId = {prod.prodId}
                            prodName = {prod.prodName}
                            prodImg = {prod.prodImg}
                            prodPrice = {prod.prodPrice} />;
            })}

        </div>
    )
}

const mapStateToProps = state => {
    return {
        products: state.products.products
    }
}

const mapDispatchToProps = dispatch => {
    return {
        fetchProducts: () => dispatch(getAllProductsStart())
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(ProdList);