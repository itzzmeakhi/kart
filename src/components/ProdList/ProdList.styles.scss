.prodlist {
    width: 90%;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    margin: 20px auto;
}