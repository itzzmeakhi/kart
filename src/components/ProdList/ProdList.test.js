import React from 'react';
import { shallow } from 'enzyme';

import ProdList from './ProdList.component';
import { storeFactory } from './../../test.utils';


const setup = (initialState={}) => {
    const store = storeFactory(initialState);
    const wrapper = shallow(<ProdList store = {store} />).dive();
    return wrapper;
}


describe('<ProdList />', () => {
    let wrapper;
    beforeEach(() => {
        const state = {
            products: {
                products : [
                    {
                        prodId: 'prod1',
                        prodName: 'abc',
                        prodImg: 'https://www.google.com',
                        prodPrice: 299
                    },
                    {
                        prodId: 'prod2',
                        prodName: 'abcd',
                        prodImg: 'https://www.google.com',
                        prodPrice: 2999
                    }
                ]
            }
        }
        wrapper = setup(state);
    })

    it('should component rendered without errors', () => {
        expect(wrapper.length).toBe(1);
    })

    it('should have valid props passed', () => {
        const products = [
            {
                prodId: 'prod1',
                prodName: 'abc',
                prodImg: 'https://www.google.com',
                prodPrice: 299
            },
            {
                prodId: 'prod2',
                prodName: 'abcd',
                prodImg: 'https://www.google.com',
                prodPrice: 2999
            }
        ]
        expect(wrapper.props().products).toEqual(products);
    })

    it('should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })

})