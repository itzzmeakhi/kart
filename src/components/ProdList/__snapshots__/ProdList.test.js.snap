// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProdList /> should match the snapshot 1`] = `ShallowWrapper {}`;
